packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+
+## Espressive CRUD Table
+
+Given the nature of our actual tables and the operations we need to perform on the data entities, a CRUD Table starts to make sense. A kind of table that not only displays data in a nice tabular way, but also allow for a richier interaction with the data.
+
+It is true that CRUD operations are needed in less scenarios, so the CRUD feature must be easily enabled and must not interfere with the regular table logic.
+
+## API
+
+The API has been divided into
+
+### Events
+
+In order to support CRUD operations, we must define a mechanism to communicate the parent any changes to the data.
+
+- onCreate
+- onUpdate
+- onDelete
+- onCustomAction
+
+Please note that the names could have been something like `onItemCreate` for semantics, but let's remember that in the future we plan to support selection and custom actions on multiple items. Our proposal is to keep these as abstract as possible.
+
+| Event          | Signature                                                  | Description                                                     |
+| -------------- | ---------------------------------------------------------- | --------------------------------------------------------------- |
+| onCreate       | `(Object: newItem) => {}`                                  | Fired when the User saves a new item                            |
+| onUpdate       | `(Array[Object]: updatedItems) => {}`                      | Fired when one or more rows have been updated by the User       |
+| onDelete       | `(Array[String]: selectedItems) => {}`                     | Fired when one or more rows have been deleted by the User       |
+| onCustomAction | `(String: actionType, Array[String]: selectedItems) => {}` | Fired when a custom action has been applied to one or more rows |
+
+### Config
+
+There will be use cases where features like multi-selection, CRUD, custom actions, etc. won't be required at all, they must be disabled by thefault but easily activated. It makes sense to have boolean props that controll such features.
+
+Another important aspect of configuration is the definition of columns, hence a complete section is devoted to it.
+
+| Name    | Type          | default | Description                                        |
+| ------- | ------------- | ------- | -------------------------------------------------- |
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
+| Create  | Boolean       | `FALSE` | If true, activates the creation of new items       |
+| Delete  | Boolean       | `FALSE` | If true, activates the deletion of items           |
+| Update  | Boolean       | `FALSE` | If true, activates the edit mode                   |
```

Comment:
The description is not very clear on the meaning of `bulk` in this document: instead of `bulk`, I should have used `multiple` to specify if multi-item-selection is enabled because that's what I ment. 

My intention was to have a JSX like this:
`
<Table bulk create update delete />
`

In the above example, `bulk`, `edit`, `create`, and `update` are boolean props that control the `multiple item selection`, `inline creation`, `inline edition`, and `deletion` features. The above example is incomplete though, because if we as developers activate those features, we would also need to consume the events they generate.

A detailed example of what's in my mind is the following:
`
<Table
  // config
  bulk
  create
  delete
  update

  coluns=[{}]

  // events
  onCreate={this.handleNewItem}
  onDelete={this.handleItemDelete}
  onUpdate={this.handleItemUpdate}
  onTrigger={this.handleTriggers}
  
  // triggers
  triggers={[
    { id: 'cancel', callback: this.handleItemCancel, ...additionalConfig },
    { id: 'view', callback: this.handleItemView, ...additionalConfig },
  ]}

  // data
  data={[{}]}
/>
`

The actions you mention are to be passed as `triggers`, the reason I choose that term is for clarity - triggers are pieces of external logic that is `triggered` by our table.

Reply:
@lordpixel I think this test example is actually something we should also include in the spec. Maybe somewhere towards the bottom. Seems like we are pretty close in terms of honing the props and surface area. We should probably have an area in the doc like this that shows how we think these might work too. This way we can also change it. 

- [ ] Could we add this to the bottom of the doc somewhere and show how some of these might look?

Reply:
Added!